marketplace_builder/views/pages/tutorials/custom-model-types/loading-related-models.liquid

```diff
@@ -0,0 +1,217 @@
+        {% query_graph 'company', id: programmer.properties.company_id, result_name: 'company' %}
+        <td> {{ company.companies.results.first.properties.name }} </td>
+        <td> {{ company.companies.results.first.properties.url }} </td>
+      {% endif  %}
+</tr>
+{% endfor %}
+{%endraw%}
+```
+
+### Step 3: Test approach
+
+Test the approach described in step 2. The results of [Measuring Execution Time of Liquid Code Fragments (time_diff)](/tutorials/performance/measuring-execution-time-liquid-time-diff) in this case will be as follows:
+
+|               | graphql |   render |
+|---------------|---------|----------|
+| benchmark-n+1 |  24.065 | 1748.686 |
+|               |  29.663 | 1862.711 |
+|               |  27.478 | 1654.610 |
+|               |  25.765 | 1787.026 |
+||—|—|
+| | 27.478 | 1787.026 |
+
+The results show that this approach produces slow output. The n+1 queries slow down the process, because displaying each record produces a request for another model. 
+
+### Step 4: Request data from related model within GraphQL query (recommended)
+
+To avoid n+1 queries, request company data within the `programmers` GraphQL query. Load related data at once along with the programmer collection using new GraphQL field called `model`.
+
+```graphql
+   model(join_on_property: "company_id") { properties }
+```
+
+`join_on_property` argument is required and is used as a foreign key of the company collection. In SQL language this could look similarly:
+
+```sql
+select * from programmers joins companies on (programmers.company_id = companies.id)
+```
+
+and after updating your graphql query looks like this now:
+
+```graphql
+query programmers {
+  programmers: customizations(name: "programmer", per_page: 200){
+    total_entries
+    results {
+      properties
+      company: model(join_on_property: "company_id") {
+        url: property(name: "url")
+        properties
+      }
+    }
+  }
+}
+```
+
+Update the `programmers/index` page. The page looks much simpler now:
+
+```liquid
+{% raw %}
+  {% for programmer in g.programmers.results %}
+    <tr>
+      <td> {{ programmer.properties.email }} </td>
+      <td> {{ programmer.properties.title }} </td>
+      <td>{{ programmer.company.properties.name }} </td>
+      <td>{{ programmer.company.properties.url }} </td>
+    </tr>
+ {% endfor %}
+{% endraw %}
+```
+
+### Step 5: Test and compare
+
+Test the second approach using the same measurement method. The results: 
+
+|                 |   graphql |  render |
+|---------------|---------|----------|
+| benchmark-model |     57.42 |  165.78 |
+|                 |     61.52 |  187.39 |
+|                 |    51.407 | 154.171 |
+|                 |    49.946 | 153.696 |
+|                 |    48.501 | 241.319 |
+|                 |    45.854 | 192.571 |
+||—|—|
+| | 64.281001 | 182.718 |
+
+Requesting data from the related model within the GraphQL query delivers results 10 times faster. 
```

Comment:
Looking at the benchmark numbers it looks like graphql query is avg 64 when benchmark n+1 is avg 27 - but render is much quicker - so i think numbers got mixed up somehow.

Reply:
@godot Could you please verify the numbers and if they are correct advise on the rewording? Thank you!